From e20da4a21dcd420b5dc7cd786fcc0c0c30a7ce44 Mon Sep 17 00:00:00 2001
From: Pete Beardmore <pete.beardmore@msn.com>
Date: Mon, 16 Jul 2018 08:46:47 +0100
Subject: [fix] hackerrank, 'validating email addresses with a filter', python
 solution

-remove '_' from second group
-ensure precise char count in final group
---
 .../validating.email.addresses.with.a.filter.py                | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/hackerrank/python.-.python.functionals.-.medium.-.validating.email.addresses.with.a.filter/validating.email.addresses.with.a.filter.py b/hackerrank/python.-.python.functionals.-.medium.-.validating.email.addresses.with.a.filter/validating.email.addresses.with.a.filter.py
index bd4029b..9cd39d4 100644
--- a/hackerrank/python.-.python.functionals.-.medium.-.validating.email.addresses.with.a.filter/validating.email.addresses.with.a.filter.py
+++ b/hackerrank/python.-.python.functionals.-.medium.-.validating.email.addresses.with.a.filter/validating.email.addresses.with.a.filter.py
@@ -2,8 +2,7 @@ import re
 
 def fun(s):
     # return True if s is a valid email, else return False
-    #return sorted(s.filter(lambda s: re.search(r'[0-9a-zA-Z_]\+@[0-9a-zA-Z_]\+\.[0-9a-zA-Z_]\+'))
-    return re.search(r'^[0-9a-zA-Z_-]+@[0-9a-zA-Z_-]+\.[0-9a-zA-Z_-]+$', s)
+    return re.search(r'^[0-9a-zA-Z_-]+@[0-9a-zA-Z]+\.[0-9a-zA-Z_-]{1,3}$', s)
 
 def filter_mail(emails):
     return list(filter(fun, emails))
-- 
2.23.0

